backend/routes/form.ts

```diff
@@ -1255,6 +1255,28 @@ async function addJobApplicationToDatabase(
 ): Promise<Responses.Id> {
     const studentId = student_id.id;
 
+    const latestJobApplication = await ormJo.getLatestJobApplicationOfStudent(
+        studentId
+    );
+
+    if (
+        latestJobApplication !== null &&
+        latestJobApplication.osoc_id === formResponse.osocId
+    ) {
+        await ormAppRo.deleteAppliedRolesByJobApplication(
+            latestJobApplication.job_application_id
+        );
+        await ormAtt.deleteAllAttachmentsForApplication(
+            latestJobApplication.job_application_id
+        );
+        await ormJoSk.deleteSkillsByJobApplicationId(
+            latestJobApplication.job_application_id
+        );
+        await ormJo.deleteJobApplication(
+            latestJobApplication.job_application_id
```

Comment:
for efficiency reasons you could use `await Promise.all` for all the queries that don't depend on each other.
This means you only have to await the `deleteJobApplication` extra.

small example (this is actually how it should be here):
```ts
await Promise.all([ormAppRo.deleteAppliedRolesByJobApplication(
              latestJobApplication.job_application_id
          ), ormAtt.deleteAllAttachmentsForApplication(
              latestJobApplication.job_application_id
          ), ormJoSk.deleteSkillsByJobApplicationId(
              latestJobApplication.job_application_id
          )
];
await ormJo.deleteJobApplication(
              latestJobApplication.job_application_id
);
```